alephium/contracts/token_bridge/token_bridge_factory.ral

```diff
@@ -1,37 +1,31 @@
 Contract TokenBridgeFactory(
-  wrappedAlphPoolTemplateId: ByteVec,      // used to create contract by copyCreateContract
   localTokenPoolTemplateId: ByteVec,       // used to create contract by copyCreateContract
   remoteTokenPoolTemplateId: ByteVec,      // used to create contract by copyCreateContract
   tokenBridgeForChainTemplateId: ByteVec,  // used to create contract by copyCreateContract
   attestTokenHandlerTemplateId: ByteVec,   // used to create contract by copyCreateContract
   unexecutedSequenceTemplateId: ByteVec    // used to create contract by copyCreateContract
 ) {
-  @using(updateFields = false)
-  pub fn getWrappedAlphPoolTemplateId() -> ByteVec {
-    return wrappedAlphPoolTemplateId
-  }
-
-  @using(updateFields = false)
+  @using(externalCallCheck = false)
   pub fn getLocalTokenPoolTemplateId() -> ByteVec {
     return localTokenPoolTemplateId
   }
 
-  @using(updateFields = false)
+  @using(externalCallCheck = false)
   pub fn getRemoteTokenPoolTemplateId() -> ByteVec {
     return remoteTokenPoolTemplateId
   }
 
-  @using(updateFields = false)
+  @using(externalCallCheck = false)
   pub fn getTokenBridgeForChainTemplateId() -> ByteVec {
     return tokenBridgeForChainTemplateId
   }
 
-  @using(updateFields = false)
+  @using(externalCallCheck = false)
   pub fn getAttestTokenHandlerTemplateId() -> ByteVec {
     return attestTokenHandlerTemplateId
   }
 
-  @using(updateFields = false)
+  @using(externalCallCheck = false)
```

Comment:
Yes, we have this rule before in [this PR](https://github.com/alephium/dev-alephium/pull/779), the compiler will skip checking external caller if the function does not modify fields and does not use assets, but we removed this rule because we think it is a bit complex after discussion.